test/python/basicaer/test_qasm_simulator.py

```diff
@@ -40,6 +50,31 @@ def setUp(self):
         transpiled_circuit.name = 'test'
         transpiled_circuit = transpile(transpiled_circuit, backend=self.backend)
         self.qobj = assemble(transpiled_circuit, shots=1000)
+        logger = getLogger()
+        logger.setLevel('DEBUG')
+        self.output = io.StringIO()
+        logger.addHandler(StreamHandlerRaiseException(self.output))
+
+    def assertExecuteLog(self, log_msg):
+        """ Runs execute and check for logs containing specified message"""
+        shots = 100
+        qr = QuantumRegister(2, 'qr')
+        cr = ClassicalRegister(4, 'cr')
+        circuit = QuantumCircuit(qr, cr)
+        execute(
+            circuit,
+            backend=self.backend,
+            shots=shots,
+            seed_simulator=self.seed)
+        self.output.seek(0)
+        # Filter unrelated log lines
+        output_lines = self.output.readlines()
+        execute_log_lines = [x for x in output_lines if x.__contains__(log_msg)]
+        self.assertTrue(len(execute_log_lines) > 0)
```

Comment:
This is good as is, but in the future,  https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs might be simpler (and should automatically handle cleaning up any state in the logger).

Reply:
Nice suggestion! Will look into it next time